internal/lms/author/service.go

```diff
@@ -0,0 +1,209 @@
+package lms
+
+import (
+	"Demo/internal/entities/author"
+	"Demo/internal/entities/book"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func CreateAuthor(db *gorm.DB) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var author author.Author
+		err := c.ShouldBindJSON(&author)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request payload"})
+			return
+		}
+
+		parsedInteger, err := strconv.ParseInt(author.DateOfBirth, 10, 64)
+		if err != nil {
+			fmt.Println("Error parsing integer:", err)
+			return
+		}
+
+		now := time.Now()
+		author.CreatedAt = fmt.Sprintf("%d-%02d-%02d %02d:%02d:%02d",
+			now.Year(),
+			now.Month(),
+			now.Day(),
+			now.Hour(),
+			now.Minute(),
+			now.Second())
+
+		author.UpdatedAt = author.CreatedAt
+
+		epochTimeSeconds := parsedInteger
+
+		epochTime := time.Unix(epochTimeSeconds, 0)
+
+		epochDateString := epochTime.Format("2006-01-02")
+		author.DateOfBirth = epochDateString
+
+		result := db.Create(&author)
+		if result.Error != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to create a new author"})
+			return
+		}
+		if author.AuthorName == "" {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Fill the author name"})
```

Comment:
create constants for error codes. error_code.go

Reply:
eg: const FillAuthorName = "Fill the author name"

Reply:
Done